src/views/Project/index.jsx

```diff
@@ -0,0 +1,247 @@
+    }, {});
+    const tagRepoList = Object.entries(tagRepoDictionary).map(
+      ([key, value]) => ({
+        label: key,
+        query: value.join(' '),
+      })
+    );
+    // list of product with no component specified for Bugzilla Query
+    const productList = projectInfo.products.filter(
+      product => typeof product === 'string'
+    );
+    // list of product with its component specified for Bugzilla Query
+    const productTemp = projectInfo.products
+      .filter(product => typeof product !== 'string')
+      .reduce((prev, cur) => ({ ...prev, ...cur }), {});
+    const productComponentList = Object.entries(productTemp).map(
+      ([key, value]) => ({
+        products: [key],
+        components: value,
+      })
+    );
+
+    this.state = {
+      projectInfo,
+      tagRepoList,
+      error: false,
+      loading: true,
+      data: [],
+      productList,
+      productComponentList,
+    };
+  }
+  componentDidMount() {
+    const { tagRepoList } = this.state;
+
+    this.fetchGithubDataNext(tagRepoList);
+    this.fetchBugzillaDataNext();
+  }
+
+  fetchGithubDataNext(tagRepoList) {
+    const { client } = this.props;
+    let allQuery = '';
+
+    if (tagRepoList.length === 0) return;
+
+    tagRepoList.forEach((tag, idx) => {
+      const noCursorQuery = `_${idx}: search(first:100, type:ISSUE, query:"${
+        tag.query
+      }")\n
+      {
+      ...Issues
+      }\n`;
+
+      allQuery = allQuery.concat(noCursorQuery);
+    });
+
+    client
+      .query({ query: gql`{${allQuery}}\n${Issues}` })
+      .catch(
+        () =>
+          new Promise(resolve => {
+            resolve(false);
+          })
+      )
+      .then(({ data, error, loading }) => {
+        const repositoriesData = Object.entries(data).map(([key, value]) => ({
+          ...value,
+          ...tagRepoList[parseInt(key.split('_')[1], 10)],
+        }));
+        // issueData is formatted like data in the state
+        const issuesData = repositoriesData.reduce(
+          (previous, repoData) => [
+            ...previous,
+            ...repoData.nodes.map(issue => {
+              const obj = {
+                project: issue.repository.name,
+                id: issue.number,
+                description: `${issue.number} - ${issue.title}`,
+                tag: issue.labels.nodes.map(node => node.name).join(','),
+                lastupdate: issue.updatedAt,
+                assignedto: issue.assignees.nodes[0]
+                  ? issue.assignees.nodes[0].login
+                  : 'None',
+              };
+
+              return obj;
+            }),
+          ],
+          []
+        );
+
+        this.setState({
+          data: _.uniqBy(
+            [...this.state.data, ...issuesData],
+            'description'
+          ).sort((a, b) => (a.lastupdate > b.lastupdate ? -1 : 1)),
+          error,
+          loading,
+        });
+      });
+  }
+
+  fetchBugs(variables) {
+    const { client } = this.props;
+
+    client
```

Comment:
Write this using an `async` function and `await` these calls instead of raw promises.

Reply:
Also, we shouldn't need to call the query API directly since we can use the `graphql` decorator. See taskcluster-web for usage.